### PR TITLE
ability to pass no-cache flag when building image

### PR DESCRIPTION
**Pull request by mattnworb**
Add configuration parameter so that the `no-cache` flag (i.e. `nocache`
in the Remote API) can be passed when building images.

### CODE REVIEW

**Comment by mattnworb** [comment]
also added test for `pullOnBuild`

**Comment by mattnworb** [comment]
@gimaker @davidxia

File: src/test/resources/pom-build-no-cache.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.spotify</groupId>
+                <artifactId>docker-maven-plugin</artifactId>
+                <version>0.1-SNAPSHOT</version>
+                <configuration>
+                    <!-- we have to supply values for all params because unit testing framework -->
+                    <!-- doesn't set default values even though the are set in mojo-->
```
Comment (davidxia) [inline src/test/resources/pom-build-no-cache.xml]: s/the/they/

same for below file

**Comment by davidxia** [comment]
:+1:

**Pull request merged**